src/chum.cpp

```diff
@@ -450,3 +454,86 @@ void Chum::setStatus(QString status) {
     m_status = status;
     emit statusChanged();
 }
+
+void Chum::getChumCache()
+{
+    if (!m_busy) {
+        m_busy = true;
+        emit busyChanged();
+    }
+
+    //% "Receiving the chum package cache"
+    setStatus(qtTrId("chum-update-cache"));
+
+    QNetworkAccessManager *netManager = new QNetworkAccessManager();
+
+    QObject::connect(netManager,&QNetworkAccessManager::finished,[=](QNetworkReply *reply) {
+        if (reply->error()){
+            qDebug() << "Unable to get package build times" << reply->errorString();
+        }
+        else {
+            m_chumPackageCache = QJsonDocument::fromJson(reply->readAll());
+            qDebug() << "Received chum package cache";
+        }
+        m_busy = false;
+        //% "Loading SailfishOS:Chum repository"
+        setStatus(qtTrId("chum-load-repositories"));
+        m_ssu.loadRepos();
+    });
+
+    // Start the network request
+    QNetworkRequest request=QNetworkRequest(QUrl(apiUrl));
+    m_busy = true;
+    emit busyChanged();
+    netManager->get(request);
+}
+
+QDateTime Chum::findPackageMTime(const QString &rpm) const
+{
+    qDebug() << Q_FUNC_INFO << rpm << Chum::instance()->repoName() << Chum::instance()->repoVersion();
+
+    QString repo = Chum::instance()->repoVersion() + "_aarch64"; //TODO need to determine device architecture
+    QString projectName = Chum::instance()->repoName().replace("-", ":");
+    QString packageName;
+    QDateTime mtime = QDateTime::fromMSecsSinceEpoch(0);
+    bool found = false;
+
+    QJsonArray projects = m_chumPackageCache.object().value("projects").toArray();
+
+    for (auto project : projects) {
+        //qDebug() << project.toObject()["name"];
+        if (project.toObject()["name"].toString() == projectName) {
+            QJsonArray repos = project.toObject()["repositories"].toArray();
+            for (auto repo : repos) {
+                //qDebug() << repo.toObject()["name"];
+                QJsonArray packages = repo.toObject()["packages"].toArray();
+                for (auto package : packages) {
+                    //qDebug() << package.toObject()["name"].toString();
+                    QJsonArray binaries = package.toObject()["binaries"].toArray();
+                    for (auto binary : binaries) {
+                        //qDebug() << binary.toString();
+                        if (binary.toString().startsWith(rpm)) {
```

Comment:
As I suggested a bit different way of doing it, it could be irrelevant. We can get false positives with libscrypt-devel and libscrypt comparison. In this case it would be OK (same package), but what if we have osmscout and osmscout-server ?